packages/docusaurus-theme-classic/src/theme/DocCard/index.tsx

```diff
@@ -104,7 +104,9 @@ function CardLink({item}: {item: PropSidebarItemLink}): JSX.Element {
       href={item.href}
       icon={icon}
       title={item.label}
-      description={doc?.description}
+      description={
+        isInternalUrl(item.href) ? doc?.description : item.description
```

Comment:
To me we should do `item.description ?? doc?.description`

Why? 
Because for anything related to the sidebar (or displaying any view related to sidebardata like this view, or the breadcrumb), the sidebar item data should always be used in priority over the doc data.

Note, this could be nice to also introduce in this PR the ability to create a custom sidebar description. It would allow rendering a different text on the index than on the doc page SEO description: more flexibility.

For example:

```js
{
      type: 'doc',
      id: 'doc1', // document ID
      label: 'Getting started',
      description: "my custom description to override docFrontMatter.description"
    },
```

We could also add a new frontMatter for it like `sidebar_description`

Does it make sense?